diff -up numactl-2.0.7/distance.c.orig numactl-2.0.7/distance.c
--- numactl-2.0.7/distance.c.orig	2012-02-15 11:38:02.521175029 +0100
+++ numactl-2.0.7/distance.c	2012-02-15 11:41:30.124189203 +0100
@@ -57,6 +57,7 @@ static int read_distance_table(void)
 	int maxnode = numa_max_node() + 1;
 	int *table = NULL;
 	int err = -1;
+	int found_nodes = 0;
 
 	for (nd = 0;; nd++) {
 		char fn[100];
@@ -85,14 +86,15 @@ static int read_distance_table(void)
 		}
 
 		parse_numbers(line, table + nd * maxnode);
+		found_nodes++;
 	}
 	free(line);
-	if (err)  {
+	if (!found_nodes)  {
 		numa_warn(W_distance,
 			  "Cannot parse distance information in sysfs: %s",
 			  strerror(errno));
 		free(table);
-		return err;
+		return -1;
 	}
 	/* Update the global table pointer.  Race window here with
 	   other threads, but in the worst case we leak one distance
