diff -up numactl-2.0.9/libnuma.c.orig numactl-2.0.9/libnuma.c
--- numactl-2.0.9/libnuma.c.orig	2014-05-22 10:59:51.062880342 +0200
+++ numactl-2.0.9/libnuma.c	2014-05-22 11:00:52.208979310 +0200
@@ -59,6 +59,8 @@ static struct bitmask *numa_memnode_ptr
 static unsigned long *node_cpu_mask_v1[NUMA_NUM_NODES];
 struct bitmask **node_cpu_mask_v2;
 
+char *nodes_allowed_list = NULL;
+
 WEAK void numa_error(char *where);
 
 #ifdef __thread
@@ -116,6 +118,8 @@ numa_fini(void)
 	FREE_AND_ZERO(numa_no_nodes_ptr);
 	FREE_AND_ZERO(numa_memnode_ptr);
 	FREE_AND_ZERO(numa_nodes_ptr);
+	if (nodes_allowed_list)
+		free(nodes_allowed_list);
 }
 
 /*
@@ -496,6 +500,12 @@ set_task_constraints(void)
 		if (strncmp(buffer,"Mems_allowed:",13) == 0) {
 			numprocnode = read_mask(mask, numa_all_nodes_ptr);
 		}
+		if (strncmp(buffer,"Mems_allowed_list:",18) == 0) {
+			nodes_allowed_list = malloc(strlen(buffer)-18);
+			strncpy(nodes_allowed_list, buffer + 19,
+				strlen(buffer) - 19);
+			nodes_allowed_list[strlen(buffer)-20] = '\0';
+		}
 	}
 	fclose(f);
 	free(buffer);
diff -up numactl-2.0.9/versions.ldscript.orig numactl-2.0.9/versions.ldscript
--- numactl-2.0.9/versions.ldscript.orig	2014-05-22 10:59:56.760889557 +0200
+++ numactl-2.0.9/versions.ldscript	2014-05-22 11:00:07.018906153 +0200
@@ -149,6 +149,7 @@ libnuma_1.2 {
     numa_tonode_memory;
     numa_tonodemask_memory;
     numa_warn;
+    nodes_allowed_list;
   local:
     *;
 } libnuma_1.1;
