diff -up numactl-2.0.7/versions.ldscript.orig numactl-2.0.7/versions.ldscript
--- numactl-2.0.7/versions.ldscript.orig	2012-10-04 19:29:27.477574120 +0200
+++ numactl-2.0.7/versions.ldscript	2012-10-04 19:30:29.817410975 +0200
@@ -153,3 +153,14 @@ libnuma_1.2 {
   local:
     *;
 } libnuma_1.1;
+
+# Symbol numa_num_possible_cpus
+# was exported in version 1.3
+
+libnuma_1.3 {
+  global:
+    numa_num_possible_cpus;
+  local:
+    *;
+} libnuma_1.2;
+
